libs/design-system/src/lib/Components/TutorialCard/PaginatedTutorialCardList.tsx

```diff
@@ -0,0 +1,55 @@
+import TutorialCard, { TutorialCardProps } from '.';
+import Pagination from '../Pagination';
+import { useEffect, useState } from 'react';
+
+export type PaginatedTutorialCardListProps = {
+  className?: string;
+
+  /**
+   * An optional identifier that, when changed, causes the pagination to
+   * be reset to the first page.
+   */
+  listId?: unknown;
+
+  pageSize?: number;
+
+  tutorials: TutorialCardProps[];
+};
+
+export const PaginatedTutorialCardList = ({
+  className,
+  listId,
+  pageSize = 4,
+  tutorials,
+}: PaginatedTutorialCardListProps) => {
+  const [page, setPage] = useState(1);
+
+  useEffect(() => {
+    // If the listId changes we reset to the first page.
+    // This allows the consuming component to force the page to reset
+    // when the list of tutorials is changed in some significant way (i.e. the
+    // list was filtered)
+    setPage(1);
+  }, [listId]);
+
+  return (
+    <div className={className}>
+      <div className="grid grid-cols-1 gap-6 md:grid-cols-4">
+        {tutorials
+          .slice((page - 1) * pageSize, page * pageSize)
+          .map((tutorialProps, index) => (
+            <TutorialCard key={index} className="w-full" {...tutorialProps} />
+          ))}
+      </div>
+      <Pagination
+        className="mt-4"
+        itemCount={tutorials.length}
+        page={page}
+        pageSize={pageSize}
+        setPage={setPage}
```

Comment:
Maybe this was planned for a future task, but it would be nice to scroll the user to the top of the section.

Currently, if the new page has less items, the user gets scrolled to the section below which is a bit confusing

Reply:
That's a great point. I'll add this. It's a little trickier because I think we probably _don't_ want this behavior when changing the filters (which also resets the pagination to the first page). It also can feel a little weird in some cases -- in particular, on mobile when the list size is reduced because of navigating to the last page. We might be able to mitigate this by animating the cards in some way when the page changes. But I'll leave that for a later enhancement.

Reply:
The scrolling looks great!